clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/orchestration/OperationsService.java

```diff
@@ -204,27 +207,59 @@ public OperationsService(
   }
 
   public List<AtomicOperation> collectAtomicOperationsFromSagas(Set<SagaId> sagaIds) {
-    if (!sagaRepository.isPresent()) {
+    if (sagaRepository.isEmpty()) {
       return Collections.emptyList();
     }
+
     // Resuming a saga-backed AtomicOperation is kind of a pain. This is because AtomicOperations
     // and their descriptions are totally decoupled from their input & description name, so we
     // have to store additional state in the Saga and then use that to reconstruct
     // AtomicOperations. It'd make sense to refactor all of this someday.
+    List<Object> seenDescriptions = new ArrayList<>();
     return sagaIds.stream()
         .map(id -> sagaRepository.get().get(id.getName(), id.getId()))
         .filter(Objects::nonNull)
         .filter(it -> !it.isComplete())
-        .map(saga -> saga.getEvent(SnapshotAtomicOperationInputCommand.class))
         .map(
-            it ->
-                convert(
-                    it.getCloudProvider(),
-                    Collections.singletonList(
-                        Collections.singletonMap(
-                            it.getDescriptionName(), it.getDescriptionInput()))))
-        .flatMap(Collection::stream)
-        .map(this::atomicOperationOrError)
+            saga ->
+                new SagaAndSnapshot(saga, saga.getEvent(SnapshotAtomicOperationInputCommand.class)))
+        .filter(
+            it -> {
+              // Reduce the list of sagas attached to the task to one for each uniquely submitted
+              // description. This is probably unnecessary long-term.
```

Comment:
Unnecessary long-term, because we'll be back to keeping a unique list of saga IDs on a task once this is rolled out, like the old behavior once was. This is really only needed for currently executing sagas that are currently appending new saga IDs.